src/FormModel.php

```diff
@@ -196,18 +196,17 @@ public function setAttribute(string $name, $value): void
         }
     }
 
-    public function processValidationResult(ResultSet $resultSet): void
+    public function processValidationResult(Result $result): void
     {
         $this->validated = false;
 
-        /** @var array<string, Resultset> $resultSet */
-        foreach ($resultSet as $attribute => $result) {
-            if ($result->isValid() === false) {
-                $this->formErrors->clear($attribute);
-                /** @psalm-suppress InvalidArgument */
-                $this->addErrors([$attribute => $result->getErrors()]);
-            }
+        $errors = $result->getErrorsIndexedByPath();
+
+        foreach ($errors as $attribute => $error) {
```

Comment:
Here can be problem in cases with nested data (if uses `Each`, `Nested` or custom rules that added errors with path).
`$errors` can be `{'attrName.0' => ['error1', 'error2']}`

Reply:
Something like this example:

Rules:
```php
final class FormWithNestedAttribute extends FormModel
{
    private ?int $id = null;
    private LoginForm $user;

    public function getRules(): array
    {
        return [
            'id' => [Required::rule()],
            'user.login' => [
                Required::rule(),
                HasLength::rule()->min(3)->tooShortMessage('Is too short.'),
            ],
        ];
    }
}
```

Test:
```php
$formModel = new FormWithNestedAttribute();
$validator = $this->createValidatorMock();
$this->assertTrue($formModel->load(['FormWithNestedAttribute' => ['user.login' => 'ad']]));
$this->assertFalse($validator->validate($formModel)->isValid());
$this->assertSame(
    ['Is too short.'],
    HtmlFormErrors::getErrorSummary($formModel, ['user.login']),
);
```

Reply:
I mean another case:

```php
public function getRules(): array
    {
        return [
            'id' => [Required::rule()],
            'names' => Each::rule(...),
        ];
    }
```

Reply:
Could you give me a real use case, of using an array in a form and validating each value.

Reply:
> Could you give me a real use case, of using an array in a form and validating each value.

Simple form with ID and array of labels.